integration/presto/src/main/java/org/apache/carbondata/presto/CarbondataModule.java

```diff
@@ -17,62 +17,150 @@
+        .as(generatedNameOf(NamenodeStats.class, connectorId));
+
+    Multibinder<HiveRecordCursorProvider> recordCursorProviderBinder =
+        newSetBinder(binder, HiveRecordCursorProvider.class);
+    recordCursorProviderBinder.addBinding().to(ParquetRecordCursorProvider.class)
+        .in(Scopes.SINGLETON);
+    recordCursorProviderBinder.addBinding().to(GenericHiveRecordCursorProvider.class)
+        .in(Scopes.SINGLETON);
+
+    binder.bind(HiveWriterStats.class).in(Scopes.SINGLETON);
+    newExporter(binder).export(HiveWriterStats.class)
+        .as(generatedNameOf(HiveWriterStats.class, connectorId));
+
+    newSetBinder(binder, EventClient.class).addBinding().to(HiveEventClient.class)
+        .in(Scopes.SINGLETON);
+    binder.bind(HivePartitionManager.class).in(Scopes.SINGLETON);
+    binder.bind(LocationService.class).to(HiveLocationService.class).in(Scopes.SINGLETON);
+    binder.bind(TableParameterCodec.class).in(Scopes.SINGLETON);
+    binder.bind(HiveMetadataFactory.class).in(Scopes.SINGLETON);
+    binder.bind(new TypeLiteral<Supplier<TransactionalMetadata>>() {
+    }).to(HiveMetadataFactory.class).in(Scopes.SINGLETON);
+    binder.bind(HiveTransactionManager.class).in(Scopes.SINGLETON);
     binder.bind(ConnectorSplitManager.class).to(CarbondataSplitManager.class).in(Scopes.SINGLETON);
+    newExporter(binder).export(ConnectorSplitManager.class)
+        .as(generatedNameOf(HiveSplitManager.class, connectorId));
     binder.bind(ConnectorPageSourceProvider.class).to(CarbondataPageSourceProvider.class)
         .in(Scopes.SINGLETON);
-    binder.bind(CarbondataHandleResolver.class).in(Scopes.SINGLETON);
-    configBinder(binder).bindConfig(CarbonTableConfig.class);
-  }
+    binder.bind(ConnectorPageSinkProvider.class).to(HivePageSinkProvider.class)
+        .in(Scopes.SINGLETON);
+    binder.bind(ConnectorNodePartitioningProvider.class).to(HiveNodePartitioningProvider.class)
+        .in(Scopes.SINGLETON);
+
+    jsonCodecBinder(binder).bindJsonCodec(PartitionUpdate.class);
 
-  public static final class TypeDeserializer extends FromStringDeserializer<Type> {
-    private final TypeManager typeManager;
+    binder.bind(FileFormatDataSourceStats.class).in(Scopes.SINGLETON);
+    newExporter(binder).export(FileFormatDataSourceStats.class)
+        .as(generatedNameOf(FileFormatDataSourceStats.class, connectorId));
 
-    @Inject public TypeDeserializer(TypeManager typeManager) {
-      super(Type.class);
-      this.typeManager = requireNonNull(typeManager, "typeManager is null");
-    }
+    Multibinder<HivePageSourceFactory> pageSourceFactoryBinder =
+        newSetBinder(binder, HivePageSourceFactory.class);
+    pageSourceFactoryBinder.addBinding().to(OrcPageSourceFactory.class).in(Scopes.SINGLETON);
+    pageSourceFactoryBinder.addBinding().to(DwrfPageSourceFactory.class).in(Scopes.SINGLETON);
+    pageSourceFactoryBinder.addBinding().to(ParquetPageSourceFactory.class).in(Scopes.SINGLETON);
```

Comment:
why orc and parquet PageSource are used here ? can add comment about this ?

Reply:
It is needed as it can read all formats from carbon connector itself